nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ScanAttribute.java

```diff
@@ -207,36 +253,62 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
 
-    private boolean allMatch(final FlowFile flowFile, final Pattern attributePattern, final Set<String> dictionary) {
-        for (final Map.Entry<String, String> entry : flowFile.getAttributes().entrySet()) {
-            if (attributePattern == null || attributePattern.matcher(entry.getKey()).matches()) {
-                if (!dictionary.contains(entry.getValue())) {
-                    return false;
+    private Map<String,String> matchAny(final FlowFile flowFile, final Pattern attributePattern, final Map<String,String> dictionary) {
+        Map<String,String> dictionaryTermMatches = new HashMap<String,String>();
+        attributeNameMatches = new HashSet<String>();
+
+        int hitCounter = 0;
+
+        for (final Map.Entry<String, String> attribute : flowFile.getAttributes().entrySet()) {
+            if (attributePattern == null || attributePattern.matcher(attribute.getKey()).matches()) {
+                attributeNameMatches.add(attribute.getKey());
+
+                if (dictionary.containsKey(attribute.getValue())) {
+                    hitCounter = setDictionaryTermMatch(dictionary, dictionaryTermMatches, hitCounter, attribute);
                 }
             }
         }
-
-        return true;
+        return dictionaryTermMatches;
     }
 
-    private boolean anyMatch(final FlowFile flowFile, final Pattern attributePattern, final Set<String> dictionary) {
-        for (final Map.Entry<String, String> entry : flowFile.getAttributes().entrySet()) {
-            if (attributePattern == null || attributePattern.matcher(entry.getKey()).matches()) {
-                if (dictionary.contains(entry.getValue())) {
-                    return true;
+    private Map<String,String> matchAll(final FlowFile flowFile, final Pattern attributePattern, final Map<String,String> dictionary) {
+        Map<String,String> dictionaryTermMatches = new HashMap<String,String>();
+        attributeNameMatches = new HashSet<String>();
+
+        int hitCounter = 0;
+
+        for (final Map.Entry<String, String> attribute : flowFile.getAttributes().entrySet()) {
+            if (attributePattern == null || attributePattern.matcher(attribute.getKey()).matches()) {
+                attributeNameMatches.add(attribute.getKey());
+
+                if (dictionary.containsKey(attribute.getValue())) {
+                    hitCounter = setDictionaryTermMatch(dictionary, dictionaryTermMatches, hitCounter, attribute);
+                } else {
+                    //if one attribute value is not found in the dictionary then no need to continue since this is a matchAll scenario.
+                    dictionaryTermMatches.clear();
+                    break;
                 }
             }
         }
+        return dictionaryTermMatches;
+    }
 
-        return false;
+    private int setDictionaryTermMatch(Map<String, String> dictionary, Map<String, String> dictionaryTermMatches, int hitCounter, Map.Entry<String, String> attribute) {
+        hitCounter++;
```

Comment:
this line should be removed.  Return type should probably be void.  The argument 'hitCounter' and all others should be final.